lib/api/objectCopy.js

```diff
@@ -0,0 +1,306 @@
+import async from 'async';
+import { errors } from 'arsenal';
+
+import data from '../data/wrapper';
+import kms from '../kms/wrapper';
+import { logger } from '../utilities/logger';
+import services from '../services';
+import utils from '../utils';
+import validateHeaders from '../utilities/validateHeaders';
+
+/**
+ * Preps metadata to be saved (based on copy or replace request header)
+ * @param {object} sourceObjMD - object md of source object
+ * @param {object} headers - request headers
+ * @param {boolean} sourceIsDestination - whether or not source is same as
+ * destination
+ * @param {AuthInfo} authInfo - authInfo from Vault
+ * @param {string} objectKey - destination key name
+ * @param {object} log - logger object
+ * @return {object} storeMetadataParams or an error
+ */
+function _prepMetadata(sourceObjMD, headers, sourceIsDestination, authInfo,
+    objectKey, log) {
+    let whichMetadata = headers['x-amz-metadata-directive'];
+    // Default is COPY
+    whichMetadata = whichMetadata === undefined ? 'COPY' : whichMetadata;
+    if (whichMetadata !== 'COPY' && whichMetadata !== 'REPLACE') {
+        return { error: errors.InvalidArgument };
+    }
+    const overrideMetadata = {};
+    if (headers['x-amz-server-side-encryption']) {
```

Comment:
I check the length of the keys of overrideMetadata below to see if any of these have been added so I can't assign these here if no data.